### PR TITLE
Openrefine Suggest API and SameAs

### PR DESCRIPTION
**Pull request by deternitydx**
I know I need more comments in the ORConstellationMapper, but I'd like to get your thoughts.

### CODE REVIEW

File: src/snac/client/openrefine/ORConstellationMapper.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    public function vocabStringLookup($type, $term) {
+        $ask = [
+            "command" => "vocabulary",
+            "query_string" => $term,
+            "type" => $type
+        ];
+        $response = $this->connect->query($ask);
+
+        if (isset($this->vocabCache[$type."|".$term]))
+            return $this->vocabCache[$type."|".$term];
```
Comment (glassjoseph) [inline src/snac/client/openrefine/ORConstellationMapper.php:104]: Wouldn't we want to check for a cache hit before sending a query?

**Comment by glassjoseph** [comment]
Yeah, I agree, maybe a few comments to explain the OR-centric terminology ('pid' = property id, 'v' = value, etc).

It looks good, and seems consistent with what little I've read of the OR Reconciliation API.  

One thing that feels a little strange to me to be querying for the sameAs vocab term and mapping together a constellation rather than simply providing the inputs (name string, sameAs URL) to the ReconciliationEngine, and asking it to query the API.

On the other hand, if we do want to extend it to support dates, subjects, or similar things in the future, it'll need to be organized somehow, might as well be a constellation. Also the ReconcilliationEngine already expects a constellation.

File: src/snac/client/openrefine/ORConstellationMapper.php (hunk 1 above)
Comment (deternitydx) [reply src/snac/client/openrefine/ORConstellationMapper.php:104]: Oh wow, I can't believe I did that!

**Comment by deternitydx** [comment]
Since the OpenRefine endpoint is just a client to snac, it's using the server's `reconcile` command which expects a Constellation object as input.  As the OR endpoint was designed, it would be the interface between the OR JSON and values and snac's rest API JSON to keep the server's interface more slimmed down.

**Pull request merged**